third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -23,6 +23,230 @@ namespace {
 //===----------------------------------------------------------------------===//
 // Data type conversion utility functions
 //===----------------------------------------------------------------------===//
+template <typename FPType> struct FPTypeInfo {
+  FPTypeInfo(Location loc, ConversionPatternRewriter &rewriter,
+             TritonLLVMOpBuilder &builder)
+      : loc(loc), rewriter(rewriter), b(builder) {}
+  std::tuple<int, int, int> getFPTypeBasicInfo() {
```

Comment:
I think we are duplicating some logic already existing in `llvm::APloat`. Can you reuse `FloatType::getFloatSemantics()` and [`APFloat` APIs](https://llvm.org/doxygen/structllvm_1_1APFloatBase.html) where suitable?